udf/kv_array_append_to_json_string.sql

```diff
@@ -0,0 +1,35 @@
+/*
+
+Returns a JSON string which has the `pair` appended to the provided `input` JSON string.
+
+Example:
+  udf_kv_array_append_to_json_string('{"foo":"bar"}', [STRUCT("baz" AS key, "boo" AS value)])
+
+  '{"foo":"bar","baz":"boo"}'
+
+  udf_kv_array_append_to_json_string('{}', [STRUCT("baz" AS key, "boo" AS value)])
+
+  '{"baz": "boo"}'
+
+*/
+CREATE OR REPLACE FUNCTION udf.kv_array_append_to_json_string(input STRING, arr ANY TYPE) AS (
+  CONCAT(
+    RTRIM(input, "}"),
+    IF(input = "{}", "", ","),
```

Comment:
```suggestion
    IF(input IS NULL OR input = "{}", "{", RTRIM(input, "}") || ",")
```

We could check for the null case here, which would remove the need for an `IFNULL` check in the view.